src/AspNetCoreModuleV2/CommonLib/iapplication.h

```diff
@@ -54,3 +55,18 @@ class IAPPLICATION
         _In_  IHttpContext       *pHttpContext,
         _Out_ IREQUEST_HANDLER  **pRequestHandler) = 0;
 };
+
+struct IAPPLICATION_DELETER
+{
+    void operator ()(IAPPLICATION* application) const
+    {
+        application->DereferenceApplication();
+    }
+};
+
+template< class APPLICATION >
+std::unique_ptr<APPLICATION, IAPPLICATION_DELETER> ReferenceApplication(APPLICATION* application)
```

Comment:
Do we need to do this for out of process? Also doesn't this more logically belong as part of `Application`?

Reply:
We need it for shim